candies/cli/cli.py

```diff
@@ -0,0 +1,59 @@
+import sys
+from typing import List, Callable, Optional, Union
+
+from candies.cli.parsers import Parser, StandardParser
+
+
+class CLI:
+    """A command line interface.
+
+    Examples:
+        @cli
+        def main(a: Arg['description'],
+                 b: Arg['description'] = 'default'):
+            \"""Some help text.\"""
+            ...
+
+        if __name__ == '__main__':
+            main()
+    """
+
+    def __init__(self, main):
+        self._main = main
+
+    def __call__(self, args: List[str] = None, parser: Parser = None):
+        """Parses the specified arguments and calls the wrapped function."""
+
+        if parser is None:
```

Comment:
Is `parser = parser or StandardParser(self._main)` is a non recommended thing to do?

Reply:
In my opinion, you are totally allowed to use such a construction. However, in this particular case, I would prefer `if` over it.